File: my-code/1/main.py
# -*- coding:utf-8 -*-
# @Time     : 2021/7/23 13: 17
# @Author   : Ranshi
# @File     : 123.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        s: dict[int, int] = {}  # 将已经读取的元素放入哈希表中, 其中目标值为key, 列表下标为val.
        for i in range(len(nums)):
            if nums[i] in s:
                return [s[nums[i]], i]
            else:
                s[target - nums[i]] = i
        return []


if __name__ == "__main__":
    print(Solution().twoSum(nums=[2, 7, 11, 15], target=9))

File: my-code/1/cpp/main.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
  vector<int> twoSum(vector<int>& nums, int target)
  {
    unordered_map<int, int> hash_table;
    for (int i = 0; i < nums.size(); i++) {
      auto it = hash_table.find(target - nums[i]);
      if (it != hash_table.end()) {
        return { it->second, i };
      }
      hash_table[nums[i]] = i;
    }
    return {};
  }
};

int
main(int argc, char const* argv[])
{
  vector<int> nums = { 2, 7, 11, 15 };
  auto res = Solution().twoSum(nums, 9);
  for (int i = 0; i < res.size(); i++) {
    cout << res.at(i);
  }

  return 0;
}
